neuraxle/metaopt/auto_ml.py

```diff
@@ -439,25 +437,118 @@ class Trainer:
 
     def __init__(
             self,
-            epochs,
-            metrics=None,
-            callbacks=None,
-            print_metrics=True,
-            print_func=None
+            epochs: int,
+            scoring_callback: ScoringCallback,
+            validation_splitter: 'BaseValidationSplitter',
+            callbacks: List[BaseCallback] = None,
+            print_func: Callable = None
     ):
-        self.epochs = epochs
-        if metrics is None:
-            metrics = {}
-        self.metrics = metrics
-        self._initialize_metrics(metrics)
+        self.epochs: int = epochs
+        self.validation_split_function = validation_splitter
 
-        self.callbacks = CallbackList(callbacks)
+        if callbacks is None:
+            callbacks = []
+        callbacks: List[BaseCallback] = [scoring_callback] + callbacks
+        self.callbacks: CallbackList = CallbackList(callbacks)
 
         if print_func is None:
             print_func = print
 
         self.print_func = print_func
-        self.print_metrics = print_metrics
+
+    def train(self, pipeline: BaseStep, data_inputs, expected_outputs=None) -> Trial:
+        """
+        Train pipeline using the validation splitter.
+        Track training, and validation metrics for each epoch.
```

Comment:
Please do a "see also" here or refer using the good doc code. 
```suggestion
        Track training, and validation metrics for each epoch.
        Note: the present method is just a shortcut to using the `execute_trial` method with less boilerplate code needed. 
Refer to `execute_trial` for full flexibility
```